### PR TITLE
Short circuit planning adapters

### PR DESCRIPTION
**Pull request by rhaschke**
This is an alternative implementation of #3258, without the need to change the return types of all planning request adapters.
This yields the following output as originally intended:
```
[ INFO] ros.moveit_ros_planning: Start state appears to be in collision with respect to group panda_manipulator
[ WARN] ros.moveit_ros_planning: Unable to find a valid state nearby the start state (using jiggle fraction of 0.050000 and 100 sampling attempts).
[DEBUG] ros.moveit_core.planning_request_adapter: Fix Start State In Collision: -10
[DEBUG] ros.moveit_core.planning_request_adapter: Fix Start State Bounds: -10
[DEBUG] ros.moveit_core.planning_request_adapter: Fix Workspace Bounds: -10
[DEBUG] ros.moveit_core.planning_request_adapter: Resolve constraint frames to robot links: -10
[DEBUG] ros.moveit_core.planning_request_adapter: Add Time Parameterization: -10
[ WARN] ros.moveit_ros_planning_interface.move_group_interface: Fail: ABORTED: START_STATE_IN_COLLISION
```

### CODE REVIEW

**Comment by rhaschke** [comment]
I further improved the debug output and cleaned up redundant code for error code -> string translation.

File: moveit_core/utils/CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@ add_library(${MOVEIT_LIB_NAME}
   src/lexical_casts.cpp
   src/xmlrpc_casts.cpp
   src/message_checks.cpp
+  src/moveit_error_code.cpp
```
Comment (AndyZe) [inline moveit_core/utils/CMakeLists.txt:7]: nit: alphabetize for readability

File: moveit_core/utils/src/moveit_error_code.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#include <moveit/utils/moveit_error_code.h>
```
Comment (AndyZe) [inline moveit_core/utils/src/moveit_error_code.cpp:35]: Just noting, this diverges a bit from what was added in MoveIt2 recently. I guess that's fine.

https://github.com/ros-planning/moveit2/blob/main/moveit_core/utils/include/moveit/utils/moveit_error_code.h

File: moveit_ros/move_group/src/move_group_capability.cpp [hunk 3]
```diff
@@ -34,6 +34,7 @@
 
 /* Author: Ioan Sucan */
 
+#include <moveit/utils/moveit_error_code.h>
 #include <moveit/moveit_cpp/moveit_cpp.h>
 #include <moveit/move_group/move_group_capability.h>
 #include <moveit/robot_state/conversions.h>
```
Comment (AndyZe) [inline moveit_ros/move_group/src/move_group_capability.cpp:40]: nit: alphabetize

File: moveit_ros/planning/plan_execution/src/plan_execution.cpp [hunk 4]
```diff
@@ -34,6 +34,7 @@
 
 /* Author: Ioan Sucan */
 
+#include <moveit/utils/moveit_error_code.h>
 #include <moveit/plan_execution/plan_execution.h>
 #include <moveit/robot_state/conversions.h>
 #include <moveit/trajectory_processing/trajectory_tools.h>
```
Comment (AndyZe) [inline moveit_ros/planning/plan_execution/src/plan_execution.cpp:40]: nit: alphabetize

File: moveit_core/planning_request_adapter/src/planning_request_adapter.cpp [hunk 5]
```diff
@@ -34,6 +34,7 @@
 
 /* Author: Ioan Sucan */
 
+#include <moveit/utils/moveit_error_code.h>
 #include <moveit/planning_request_adapter/planning_request_adapter.h>
 #include <functional>
 #include <algorithm>
```
Comment (AndyZe) [inline moveit_core/planning_request_adapter/src/planning_request_adapter.cpp:40]: nit: alphabetize/sort

**Review by AndyZe — COMMENTED** [review]
First pass of comments

**Review by AndyZe — COMMENTED** [review]
I've tested the PR. It seems to work well. I'll approve after you address the minor comments.

File: moveit_ros/move_group/src/move_group_capability.cpp (hunk 3 above)
Comment (rhaschke) [reply moveit_ros/move_group/src/move_group_capability.cpp]: I shifted the newly introduced include, but I won't manually sort all existing ones.

File: moveit_core/utils/src/moveit_error_code.cpp (hunk 2 above)
Comment (rhaschke) [reply moveit_core/utils/src/moveit_error_code.cpp:35]: I explicitly decided to return `const char*` and avoid allocating new memory for std::string.

**Review by AndyZe — APPROVED** [review]

**Pull request merged**